src/assets/CollectibleDetectionController.ts

```diff
@@ -309,15 +314,22 @@ export class CollectibleDetectionController extends BaseController<
     if (!this.isMainnet() || this.disabled) {
       return;
     }
-    const requestedSelectedAddress = this.config.selectedAddress;
+    const { selectedAddress } = this.config;
+
+    let { chainId } = this.config;
+    if (typeof chainId === 'string' && isHexString(chainId)) {
+      chainId = `${parseInt(chainId, 16)}` as `${number}`;
+    } else if (typeof chainId === 'number') {
+      chainId = `${chainId}` as `${number}`;
+    }
 
     /* istanbul ignore else */
-    if (!requestedSelectedAddress) {
+    if (!selectedAddress) {
       return;
     }
 
     await safelyExecute(async () => {
-      const apiCollectibles = await this.getOwnerCollectibles();
+      const apiCollectibles = await this.getOwnerCollectibles(selectedAddress);
```

Comment:
I'm not understanding what we achieve by getting `selectedAddress` in `detectCollectibles` and passing it in to `getOwnerCollectibles` rather than just pulling `selectedAddress` off config in `getOwnerCollectibles`?

Reply:
We just want to avoid grabbing it in different places to make sure it does not change along the execution of the procedure